Orbitersdk/samples/ProjectApollo/src_lm/lm_ecs.cpp

```diff
@@ -381,20 +413,20 @@ void LEMCabinRepressValve::SystemTimestep(double simdt)
 	{
 		cabinRepressValve->flowMax = 396.0 / LBH;
 
-		if (cabinRepressCB->IsPowered() && !pressRegulatorASwitch->GetState() == 0 && !pressRegulatorBSwitch->GetState() == 0)
+		if (cabinRepressCB->IsPowered() && (pressRegulatorASwitch->GetState() != 0 || pressRegulatorBSwitch->GetState() != 0))
 		{
 			//Cabin pressure
 			double cabinpress = cabinRepressValve->out->parent->space.Press;
 
-			if (cabinpress < 4.07 / PSI && cabinRepressValve->in->open == 0)
+			if (lem->CabinPressureSwitch.GetPressureSwitch() != 0 && cabinRepressValve->in->open == 0)
 			{
 				cabinRepressValve->in->Open();
-				EmergencyCabinRepressRelay = true;
 			}
-			else if (cabinpress > 4.7 / PSI && cabinRepressValve->in->open == 1)
+			else if (lem->CabinPressureSwitch.GetPressureSwitch() == 0 && cabinRepressValve->in->open == 1)
 			{
 				cabinRepressValve->in->Close();
 			}
+			EmergencyCabinRepressRelay = true;
```

Comment:
This needs the additional check on the valve state, as I said in the IRC chat:

`if (cabinRepressValve->in->open)
{
EmergencyCabinRepressRelay = true;
}`